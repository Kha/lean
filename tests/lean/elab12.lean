#check (assume a : nat, have H : 0, from rfl a,
 (H a a) : ∀ a : nat, a = a)

#check (assume a : nat, have H : a = a, from rfl a,
 (H a a) : ∀ a : nat, a = a)

#check (assume a : nat, have H : a = a, from a + 0,
 (H a a) : ∀ a : nat, a = a)

#check (assume a : nat, have H : a = a, from rfl,
 (H a) : ∀ a : nat, a = a)

#check (assume a : nat, have H : a = a, from rfl,
 H : ∀ a : nat, a = a)
