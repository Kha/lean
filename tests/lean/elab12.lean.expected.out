elab12.lean:1:33: error: type expected at
  0
term has type
  ℕ
elab12.lean:1:41: error: function expected at
  rfl
term has type
  ?m_2 = ?m_2
Additional information:
elab12.lean:1:41: context: switched to simple application elaboration procedure because failed to use expected type to elaborate it, error message
  too many arguments
elab12.lean:2:2: error: function expected at
  H
term has type
  ⁇
λ (a : ℕ), have H : ⁇, from ⁇, ⁇ : ∀ (a : ℕ), a = a
elab12.lean:4:45: error: function expected at
  rfl
term has type
  ?m_2 = ?m_2
Additional information:
elab12.lean:4:45: context: switched to simple application elaboration procedure because failed to use expected type to elaborate it, error message
  too many arguments
elab12.lean:5:2: error: function expected at
  H
term has type
  a = a
λ (a : ℕ), have H : a = a, from ⁇, ⁇ : ∀ (a : ℕ), a = a
elab12.lean:7:47: error: invalid have-expression, term
  a + 0
has type
  ℕ
but is expected to have type
  a = a
elab12.lean:8:2: error: function expected at
  H
term has type
  a = a
λ (a : ℕ), have H : a = a, from ⁇, ⁇ : ∀ (a : ℕ), a = a
elab12.lean:11:2: error: function expected at
  H
term has type
  a = a
λ (a : ℕ), have H : a = a, from rfl, ⁇ : ∀ (a : ℕ), a = a
λ (a : ℕ), have H : a = a, from rfl, H : ∀ (a : ℕ), a = a
